lib/rules/no-dom-import.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+const DOM_TESTING_LIBRARY_MODULES = [
+  'dom-testing-library',
+  '@testing-library/dom',
+];
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow importing from DOM Testing Library',
+      category: 'Best Practices',
+      recommended: false,
+      url: getDocsUrl('no-dom-import'),
+    },
+    messages: {
+      noDomImport:
+        'import from DOM Testing Library is restricted, import from corresponding Testing library framework instead',
+    },
+    fixable: null,
```

Comment:
It would be awesome (and I think pretty easy though I'm not sure of that) to include a fix for this rule. I can look into it if you want.

Reply:
It should, and I was thinking about it but I prefer to leave the fixable for a future version. Every time I think about this one it seems more complex to implement as you don't know which TL framework the query must be imported (don't think about react only, this works for vue and angular). So for fixing the rule we could try to guess from where the user should import it by looking for other "@testing-library/<suffix>" text and get that suffix to replace it by "@testing-library/dom" but then... what if it's the first import in the file referring to testing library? We can't guess from where we have to reimport. Did you have something else in mind for this?

Reply:
Oh yes that's true. This is the same problem than above, I have two potential solutions 👆